src/pb_tensor.cc

```diff
@@ -305,12 +348,42 @@ PbTensor::Memory()
 
 #ifdef TRITON_PB_STUB
 std::shared_ptr<PbTensor>
-PbTensor::FromDLPack(const std::string& name, const py::capsule& dlpack_tensor)
+PbTensor::FromDLPack(const std::string& name, const py::object& tensor)
 {
   if (name == "") {
     throw PythonBackendException("Tensor name cannot be an empty string.");
   }
+  if (py::isinstance<py::capsule>(tensor)) {
+    return FromDLPackCapsule(name, tensor);
+  } else if (py::hasattr(tensor, "__dlpack__")) {
```

Comment:
Do we need to make sure it implements both `__dlpack__` and `__dlpack_device__`? It looks like we don't need `__dlpack_device__` when python backend is the consumer since the same information is available in dlpack capsule?

Reply:
We don't need to ensure __dlpack_device__ is implemented here, since we don't call it explicitly, and it is true, capsule should contain device info.